app/uk/gov/hmrc/tai/viewModels/IncomeSourceComparisonViewModel.scala

```diff
@@ -60,13 +59,24 @@ object IncomeSourceComparisonViewModel extends ViewModelHelper with TaxAccountFi
   }
 
   private def incomeSourceDetail(taxCodeIncomes: Seq[TaxCodeIncome], employments: Seq[Employment], taxYearStatus:String): Seq[IncomeSourceDetail] = {
-    taxCodeIncomes.flatMap { (t: TaxCodeIncome) =>
-      t.employmentId.flatMap { (id: Int) =>
-        employments.find(_.sequenceNumber == id).map{ e:Employment =>
-          val amount = withPoundPrefixAndSign(MoneyPounds(t.amount, 0))
-          IncomeSourceDetail(e.name,e.sequenceNumber, amount, taxYearStatus)
+    taxYearStatus match {
+      case TaiConstants.CurrentTaxYearPlusOne =>
+        taxCodeIncomes.map { taxCodeIncome =>
+          lazy val amount = withPoundPrefixAndSign(MoneyPounds(taxCodeIncome.amount, 0))
+          taxCodeIncome.employmentId match {
+            case Some(id) => IncomeSourceDetail(taxCodeIncome.name, id, amount, taxYearStatus)
+            case _ => throw new RuntimeException("Employment id is missing")
```

Comment:
Could this return an empty instead of throwing?

Reply:
It is an exceptional condition that should not really happen hence the exception. Do you mean an empty IncomeSourceDetail? Happy to discuss.